x-pack/plugins/security/common/model/authenticated_user.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { User } from './user';
+
+const REALMS_ELIGIBLE_FOR_PASSWORD_CHANGE = ['reserved', 'native'];
+
+interface UserRealm {
+  name: string;
+  type: string;
+}
+
+/**
+ * Represents the currently authenticated user.
+ */
+export interface AuthenticatedUser extends User {
+  /**
+   * The name and type of the Realm that has authenticated the user.
+   */
+  authentication_realm: UserRealm;
+
+  /**
+   * The name and type of the Realm where the user information were retrieved from.
+   */
+  lookup_realm: UserRealm;
+}
+
+export function canUserChangePassword(user: AuthenticatedUser) {
```

Comment:
__note:__ if I read code correctly this method can only be used with `AuthenticatedUser` from `ChangePassword` component (not `ChangePasswordForm`) that is rendered only within `AccountManagementPage` that is rendered only for currently authenticated user. That's why I moved this method here.

Reply:
that's absolutely right!

Reply:
Awesome, thanks!